docs/setup.md

```diff
@@ -50,16 +64,93 @@ To get help regarding the usage of service account setup command within the snap
+```
+
+The above command sets up a service account for user ```demonamespace:demouser``` for Spark job submission using configuration properties coming from the specified 
+properties file while overriding the configuration properties ```spark.app.name``` and ```spark.executor.instances```.
+
+As you will discover further in the documentation for [job submission](/docs/submit.md), this service account along with it's default configuration properties can be used to submit a Spark job.
+
+For example, assuming the properties file provided has configuration details to access data in S3, one could submit a job like
+```bash
+spark-client.spark-submit  --username demouser --namespace demonamespace --deploy-mode cluster --conf spark.app.name=demo-spark-s3-app $S3_PATH_FOR_CODE_FILE
+```
+This would launch the spark job with configuration coming from the service account for user ```demonamespace:demouser``` but the app name would be ```demo-spark-s3-app```. 
+
+**_NOTE:_** The comamnd described above does not create a kubernetes namespace but needs it to be there. It does however create the requested username in the specified and existing namespace. 
+
+#### Creating configuration for a Service Account after account creation
+
+The configuration functionality mentioned in the previous section can be done after account creation as well. Useful for redefining the configuration associated with a existing service account.
+
+```bash
+spark-client.setup-spark-k8s --username demouser --namespace demonamespace sa-conf-create --properties-file /home/demouser/conf/spark-overrides.conf  --conf spark.app.name=demo-spark-app-overrides
+```
+
+The above command will **drop** the existing configuration associated with service account ```demonamespace:demouser``` and recreate it with properties in the new properties file and override the ```spark.app.name```.
+
+All job submissions using this service-account will use the new default configuration going forward.
+
+
+#### Inspecting configuration for a Service Account
+
+For inspecting the current properties configured for a service account, the ```sa-conf-get``` utility comes handy.
 
+```bash
+spark-client.setup-spark-k8s --username demouser --namespace demonamespace sa-conf-get --conf spark.app.name --conf spark.kubernetes.namespace --conf spark.executor.instances
+```
+
+If you wish to not pick specific properties and see all of them for service account ```demonamespace:demouser```
+
+```bash
+spark-client.setup-spark-k8s --username demouser --namespace demonamespace sa-conf-get
+```
+
+#### Deleting configuration for a Service Account
+
+For deleting all configuration associated with a service account,
+
+```bash
+spark-client.setup-spark-k8s --username demouser --namespace demonamespace sa-conf-delete
+```
+
+#### Inspecting Primary Service Account
```

Comment:
Feels like following functionalities would be useful 
-- mark an existing service account as primary
-- list all service accounts managed by Spark client
-- register an existing service account for use with Spark client (will be useful when people using kubeflow like products just want to install the client and go with the existing service account which they might have created in a very custom way which cannot be repeated with Spark client.)

Reply:
Yes, I agree. They are all very relevant points. I would put them in the backlog and see whether we will have margin to address (some or all of) them in the next sprints!